Website/admin/Skins/TreeViewMenu.ascx.cs

```diff
@@ -339,15 +339,15 @@ private void BuildTree(DNNNode objNode, bool blnPODRequest)
             {
                 if (!string.IsNullOrEmpty(Level))
                 {
-                    switch (Level.ToLower())
+                    switch (Level.ToLowerInvariant())
                     {
                         case "root":
                             break;
                         case "child":
                             blnAddUpNode = true;
                             break;
                         default:
-                            if (Level.ToLower() != "root" && PortalSettings.ActiveTab.BreadCrumbs.Count > 1)
+                            if (Level.ToLowerInvariant() != "root" && PortalSettings.ActiveTab.BreadCrumbs.Count > 1)
```

Comment:
Please use `String#Equals(String, StringComparison)`

Reply:
Hi @tpluscode , I didn't do these refactoring because we have issue in previous review, i made the refactoring and it cause hard to review, and if we want to do refactoring, we should create a new improvement task.
I think we should change everywhere when we try to compare the string, because we can't know what value will included in the string, like file extension maybe "gif", enum value maybe "Ignore" etc, so it possible to have the convert issue and we need fix it.

root cause here is when use ToLower function, in Turkish culture, it will not convert "Install" to "install", first char will be the special char which not have a dot, then when compare to "install", it will never equal, and cause compare failed.

Reply:
Why was it hard to review? Which issue was that? I shall think that the diffs will be pretty much the same. One line removed, one line added.

And I don't understand your argument. In terms of functionality, the two alternatives yield same result

```c#
Level.ToLower() != "root"
"root".Equals(Level, StringComparison.InvariantCultureIgnoreCase) == false
```

The latter is just more efficient. Given how many of those we have it could make quite a bit of a difference.

Reply:
Where do you get that string.Equals is faster than just using ==?

Your suggestion makes for less readable code. If it is significantly faster, I'd like to see some proof of that. That mens proper analysis, and presenting that case in a new github issue. If not, then let's keep it the way Ben fixed it.

Reply:
It's not about `Equals` vs equality operator but about calling `ToLower`/`ToLowerInvariant` every time to perform the comparison. Especially case like `"a".ToLower() == "A".ToLower()` 
is very bad because it there will be all those unnecessary allocations to lower-cased strings. These allocation do not happen with `String.Equals`.

So CPU cycles are not the only measurement here. There will also be a gain in memory consumption and garbage collection of all those strings. 

I see your point about seeing real numbers. Do others in @dnnsoftware/tag also share that view?